integrations/dell_idrac/dell_idrac_test.py

```diff
@@ -0,0 +1,203 @@
+             patch.object(DelliDRACClient, "check_username_by_id", side_effect=SaasException("Username mismatch with user_id")):
+            with self.assertRaises(SaasException) as ctx:
+                plugin.change_password()
+            self.assertIn("Username mismatch with user_id", str(ctx.exception))
+
+    def test_change_password_http_error(self):
+        plugin = self.plugin()
+        with patch.object(DelliDRACClient, "get_user_id_from_user_fields", return_value="42"), \
+             patch.object(DelliDRACClient, "check_username_by_id"), \
+             patch.object(DelliDRACClient, "change_dell_idrac_user_password", side_effect=SaasException("Password rotation failed: 500")):
+            with self.assertRaises(SaasException) as ctx:
+                plugin.change_password()
+            self.assertIn("Password rotation failed", str(ctx.exception))
+
+    def test_rollback_password_success(self):
+        plugin = self.plugin(prior_password=Secret("OldPassword123"))
+        plugin._client = MagicMock()
+        plugin._SaasPlugin__user_id = "42"
+        plugin.user.prior_password = Secret(("SomeOtherPassword", "OldPassword123"))
+        plugin._client.change_dell_idrac_user_password = MagicMock()
+        plugin.rollback_password()
+        plugin._client.change_dell_idrac_user_password.assert_called_once_with("42", "OldPassword123")
+
+    def test_rollback_password_no_prior(self):
+        plugin = self.plugin(prior_password=None)
+        plugin._client = MagicMock()
+        plugin._SaasPlugin__user_id = "42"
+        plugin.user.prior_password = None
+        with self.assertRaises(Exception):
+            plugin.rollback_password()
+
+    def test_rollback_password_http_error(self):
+        plugin = self.plugin(prior_password=Secret("OldPassword123"))
+        plugin._client = MagicMock()
+        plugin._SaasPlugin__user_id = "42"
+        plugin.user.prior_password = Secret(("SomeOtherPassword", "OldPassword123"))
+        plugin._client.change_dell_idrac_user_password.side_effect = SaasException("Password change request failed: 500")
+        with self.assertRaises(SaasException) as ctx:
+            plugin.rollback_password()
+        self.assertIn("Rollback failed", str(ctx.exception))
+
+
+
+class DelliDRACClientTest(unittest.TestCase):
+
+    def setUp(self):
+        user = MagicMock()
+        user.username.value = "testuser"
+        self.client = DelliDRACClient(
+            admin_username="admin",
+            admin_password="adminpass",
+            idrac_ip="1.2.3.4",
+            user=user
+        )
+
+    @patch("dell_idrac.requests.get")
+    def test_check_username_by_id_success(self, mock_get):
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = {"UserName": "testuser"}
+        mock_get.return_value = mock_response
+
+        # Should not raise
+        self.client.check_username_by_id("42")
```

Comment:
Why should'g this raise an exception. The usernames is different, 42 and testuser